core/tests/utils/temp_core.rs

```diff
@@ -0,0 +1,58 @@
+use tempfile::TempDir;
+
+use stump_core::StumpCore;
+
+pub async fn get_temp_core() -> (StumpCore, TempDir) {
+	let temp_dir = TempDir::new().unwrap();
+
+	let config_dir = temp_dir.path().to_string_lossy().to_string();
+	let config = StumpCore::init_config(config_dir).unwrap();
+	let core = StumpCore::new(config).await;
+
+	let migration_res = core.run_migrations().await;
+	assert!(
+		migration_res.is_ok(),
+		"Failed to run migrations: {:?}",
+		migration_res
+	);
```

Comment:
Just a note that IIRC this won't actually run migrations in debug/test. I _think_ it will just push onto the DB. I'd have to double check to be sure. Either way, it is needed